pkg/kubelet/nodestatus/setters.go

```diff
@@ -470,6 +472,17 @@ func ReadyCondition(
 		if len(missingCapacities) > 0 {
 			errs = append(errs, fmt.Errorf("Missing node capacity for resources: %s", strings.Join(missingCapacities, ", ")))
 		}
+		if servingCertStatusFunc != nil {
+			crt := servingCertStatusFunc()
```

Comment:
this only affects readiness if the kubelet has serving cert rotation enabled, right? are there other readiness checks that can go bad after startup like this?

Reply:
> this only affects readiness if the kubelet has serving cert rotation enabled, right?

Yes, if static serving cert is used this check will never trigger. Ideally we could also watch static cert's expiration. Is it useful for this PR?

> are there other readiness checks that can go bad after startup like this?

IIUC, yes. `runtimeErrorsFunc` and `networkErrorsFunc` rely on periodic health checks of CRI and CNI. If those start failing, node status flips.